### PR TITLE
[cmake] Ignore stringop-truncation

### PR DESCRIPTION
**Pull request by mlohvynenko**

### CODE REVIEW

File: include/aos/common/tools/error.hpp [hunk 1]
```diff
@@ -267,7 +267,10 @@ class Error {
     void CopyMessage(const char* msg)
     {
         if (msg != nullptr) {
+#pragma GCC diagnostic push
```
Comment (al1img) [inline include/aos/common/tools/error.hpp:270]: Disable this warning in main cmake file.

File: include/aos/common/tools/string.hpp [hunk 2]
```diff
@@ -496,7 +496,7 @@ class String : public Array<char> {
         Clear();
 
         // cppcheck-suppress wrongPrintfScanfArgNum
-        auto ret = snprintf(Get(), MaxSize() + 1, format, args...);
```
Comment (al1img) [inline include/aos/common/tools/string.hpp:499]: Is there any warning here? Should be done in separate commit with proper commit message.

**Review by al1img — APPROVED** [review]

**Review by MykolaSolyanko — APPROVED** [review]
